couchdb/tests/server/views.py

```diff
@@ -118,9 +118,103 @@ def test_return_nothing(self):
         views.map_doc(self.server, doc)
 
 
+class ReduceTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.server = MockQueryServer()
+
+    def test_reduce(self):
+        """should reduce map function result"""
+        state.add_fun(
+            self.server,
+            'def mapfun(doc):\n'
+            '  return ([doc["_id"], i] for i in range(10))'
+        )
+        result = views.map_doc(self.server, {'_id': 'foo'})
+        rresult = views.reduce(
+            self.server,
+            ['def reducefun(keys, values): return sum(values)'],
+            result[0]
+        )
+        self.assertEqual(rresult, [True, [45]])
+
+    def test_reduce_by_many_functions(self):
+        """should proceed map keys-values result by multiple reduce functions"""
+        state.add_fun(
+            self.server,
+            'def mapfun(doc):\n'
+            '  return ([doc["_id"], i] for i in range(10))'
+        )
+        result = views.map_doc(self.server, {'_id': 'foo'})
+        rresult = views.reduce(
+            self.server,
+            ['def reducefun(keys, values): return sum(values)',
+             'def reducefun(keys, values): return max(values)',
+             'def reducefun(keys, values): return min(values)'],
+            result[0]
+        )
+        self.assertEqual(rresult, [True, [45, 9, 0]])
```

Comment:
This test case confuse me. I have checked the query server protocol, the format of `reduct` is correct: 
`["reduce", ["fun_src1", "fun_src2", ...], [ [key, val] ...]`

But a question came to my mind: This `reduce` command imply that a view in ddoc can own more than one reduce functions? I never see that!
A ddoc always looks like this:

```javascript
{
  "_id": "_design/app",
  "views": [
    "myview": {
      "map": "...",
      "reduce": "..."
    }
  ]
}
```

So ... which situation will make couchdb send multi `reduce` functions?